phoenix-core-client/src/main/java/org/apache/phoenix/compile/StatementContext.java

```diff
@@ -111,6 +113,7 @@ public StatementContext(StatementContext context) {
         this.queryLogger = context.queryLogger;
         this.isClientSideUpsertSelect = context.isClientSideUpsertSelect;
         this.isUncoveredIndex = context.isUncoveredIndex;
+        this.hasFirstValidResult = context.hasFirstValidResult;
```

Comment:
This should be this.hasFirstValidResult = new AtomicBoolean(context.getHasFirstValidResult());